pkg/controller/daemon/daemon_controller.go

```diff
@@ -1031,7 +1035,18 @@ func (dsc *DaemonSetsController) syncNodes(ds *apps.DaemonSet, podsToDelete, nod
 
 				podTemplate := &template
 
-				if utilfeature.DefaultFeatureGate.Enabled(features.ScheduleDaemonSetPods) {
+				nodeCanUseMatchFields := false
+				if node, err := dsc.nodeLister.Get(nodesNeedingDaemonPods[ix]); err != nil {
+					glog.Errorf("unknown node %s, disabling ScheduleDaemonSetPods using MatchFields: %v", nodesNeedingDaemonPods[ix], err)
+				} else if kubeletVersion, err := utilversion.ParseSemantic(node.Status.NodeInfo.KubeletVersion); err != nil {
+					glog.Errorf("unknown kubelet version %s for node %s, disabling ScheduleDaemonSetPods using MatchFields: %v", node.Status.NodeInfo.KubeletVersion, nodesNeedingDaemonPods[ix], err)
+				} else if kubeletVersion.LessThan(matchFieldVersion) {
+					glog.V(4).Infof("kubelet version %s on node %s is less than %s, disabling ScheduleDaemonSetPods using MatchFields", node.Status.NodeInfo.KubeletVersion, nodesNeedingDaemonPods[ix], matchFieldVersion)
```

Comment:
I think, we should increase the log level(don't mind, if this V(2) or 1) for this message since we are obfuscating  from the end-user the fact that enabling `ScheduleDaemonSetPods` feature has no impact if kubelet version is less than 1.11 which may be bad user experience.

Reply:
no, a 2-version skewed kubelet should not flood logs. disabling the MatchFields-based scheduling is working properly, it's not actionable and shouldn't matter to the administrator.

Reply:
But what about in 1.10 cluster, where the admin explicitly enabled `ScheduleDaemonSetPods` and he/she notices that the cluster is not behaving as per its configuration?